src/test/java/twitterapp/src/ResourceTest.java

```diff
@@ -40,35 +56,46 @@ public void testTweetLength(){
         assertEquals("Tweet("+shortTweet+") has been posted",r.getEntity().toString());
     }
 
+    @Test
+    public void testFailTweet(){
+        try{
+            when(mockTwitter.updateStatus("test tweet")).thenThrow(new TwitterException("There was a problem on the server side, please try again later."));
+            r = noMockTwitterResource.postTweet("test tweet");
+            assertEquals(500, r.getStatus());
```

Comment:
use the Status code enum instead of 500 and compared the expected error message as well